hyperspy/drawing/utils.py

```diff
@@ -343,6 +343,466 @@ def _set_spectrum_xlabel(spectrum, ax):
+            if no_nans:
+                data = np.nan_to_num(data)
+
+            # Get handles for the signal axes and axes_manager
+            axes_manager = im.axes_manager
+            axes = axes_manager.signal_axes
+
+            if axes_manager.signal_dimension == 2:
+                # get calibration from a passed axes_manager
+                shape = axes_manager._signal_shape_in_array
+
+                # Reshape the data for input into imshow (if not rgb)
+                if not isrgb[i]:
+                    data = data.reshape(shape)
+
+            # Set dimensions of images
+            xaxis = axes[0]
+            yaxis = axes[1]
+
+            extent = (
+                xaxis.low_value,
+                xaxis.high_value,
+                yaxis.high_value,
+                yaxis.low_value,
+            )
+
+            if not isinstance(aspect, (int, long, float)) and aspect not in ['auto', 'square', 'equal']:
+                print 'Did not understand aspect ratio input. Using \'auto\' as default.'
+                aspect = 'auto'
+
+            if aspect is 'auto':
+                if float(yaxis.size) / xaxis.size < min_asp:
+                    factor = min_asp * float(xaxis.size) / yaxis.size
+                elif float(yaxis.size) / xaxis.size > min_asp ** -1:
+                    factor = min_asp ** -1 * float(xaxis.size) / yaxis.size
+                else:
+                    factor = 1
+                asp = np.abs(factor * float(xaxis.scale) / yaxis.scale)
+            elif aspect is 'square':
+                asp = abs(extent[1] - extent[0]) / abs(extent[3] - extent[2])
+            elif aspect is 'equal':
+                asp = 1
+            elif isinstance(aspect, (int, long, float)):
+                asp = aspect
+
+            # Plot image data, using vmin and vmax to set bounds, or allowing them
+            # to be set automatically if using individual colorbars
+            if colorbar is 'single' and not isrgb[i]:
+                im = ax.imshow(data,
+                               cmap=cmap, extent=extent,
+                               interpolation=interp,
+                               vmin=gl_min, vmax=gl_max,
+                               aspect=asp,
+                               *args, **kwargs)
+            else:
+                im = ax.imshow(data,
+                               cmap=cmap, extent=extent,
+                               aspect=asp,
+                               interpolation=interp,
+                               *args, **kwargs)
+
+            # Label the axes
+            if isinstance(axes[0].units, trait_base._Undefined):
+                axes[0].units = 'pixels'
+            if isinstance(axes[1].units, trait_base._Undefined):
+                axes[1].units = 'pixels'
+            if isinstance(axes[0].name, trait_base._Undefined):
+                axes[0].name = 'x'
+            if isinstance(axes[1].name, trait_base._Undefined):
+                axes[1].name = 'y'
```

Comment:
> Shouldn't you let the axis undefined if they are undefined?

If this is the case, should we just hide the axes, then? It is really ugly (in my opinion) to have big "<undefined>" labels around the plot